core/app/models/spree/product.rb

```diff
@@ -194,11 +189,7 @@ def property(property_name)
     def set_property(property_name, property_value, property_presentation = property_name)
       ActiveRecord::Base.transaction do
         # Works around spree_i18n #301
-        property = if Property.exists?(name: property_name)
-          Property.where(name: property_name).first
-        else
-          Property.create(name: property_name, presentation: property_presentation)
-        end
+        property = Property.create_with(presentation: property_presentation).find_or_create_by(name: property_name)
```

Comment:
I don't have idea how to do this line better. My code perform 1 query in best scenario and 2 when we don't have Property record.

Previous version always perform 2 queries, even if record exists.